vowpalwabbit/parse_args.cc

```diff
@@ -1278,6 +1279,7 @@ void parse_reductions(options_i& options, vw& all)
   all.reduction_stack.push(explore_eval_setup);
   all.reduction_stack.push(ExpReplay::expreplay_setup<'c', COST_SENSITIVE::cs_label>);
   all.reduction_stack.push(Search::setup);
+  all.reduction_stack.push(VW::shared_feature_merger::shared_feature_merger_setup);
```

Comment:
This may be more of a @JohnLangford question, but is this the right place in the stack for this setup to run? It seems like we would want to only run this for reductions that inject csoaa_ldf or wap_ldf, even if it not provided at the command-line, but if this reduction is above them in the stack, it will never know that this argument gets injected.

Reply:
Moved the reduction further down the stack (after the cbify/warm_cb reductions), traced and added the options that would result in calls into csoaa::do_actual_learning

Reply:
Its not a perfect solution, but it should work until we can refactor the reduction stack into something more explicit